apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/theadpool/ThreadPoolRegister.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.theadpool;
+
+import static java.util.Collections.synchronizedList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ThreadPoolExecutor;
+
+public class ThreadPoolRegister {
+
+    private static final List<ThreadPoolSourceData> THREAD_POOL_SOURCE_DATA_LIST = synchronizedList(new ArrayList<>());
+
+    public static void registerThreadPoolSource(ThreadPoolName threadPoolName, ThreadPoolExecutor threadPoolExecutor) {
+        if (THREAD_POOL_SOURCE_DATA_LIST.stream().noneMatch(it -> it.getThreadPoolName().equals(threadPoolName))) {
+            ThreadPoolSourceData threadPoolSourceData = new ThreadPoolSourceData();
+            threadPoolSourceData.setThreadPoolName(threadPoolName);
+            threadPoolSourceData.setHasBuildMetric(false);
+            threadPoolSourceData.setThreadPoolExecutor(threadPoolExecutor);
+            THREAD_POOL_SOURCE_DATA_LIST.add(threadPoolSourceData);
+        }
```

Comment:
Without `lock` block(ReentrantLock, this `synchronizedList` would not help. You still could face thread pools in the same duplicate name.